apps/condo/domains/miniapp/schema/CustomValue.js

```diff
@@ -0,0 +1,369 @@
+        },
+    },
+}
+
+const INVALID_OBJECT_ID = 'INVALID_OBJECT_ID'
+const INVALID_SOURCE = 'INVALID_SOURCE'
+const INVALID_DATA = 'INVALID_DATA'
+const INVALID_CUSTOM_FIELD = 'INVALID_CUSTOM_FIELD'
+const INVALID_UNIT = 'INVALID_UNIT'
+const INVALID_ADDRESS_KEY = 'INVALID_ADDRESS_KEY'
+const ALREADY_EXISTS_UNIQ_KEY = 'ALREADY_EXISTS_UNIQ_KEY'
+const ALREADY_EXISTS_OBJECT_ID = 'ALREADY_EXISTS_OBJECT_ID'
+
+const ERRORS = {
+    INVALID_DATA: {
+        code: BAD_USER_INPUT,
+        type: INVALID_DATA,
+        message: 'Provided data is invalid. Check type and validation rules for provided customField',
+    },
+    INVALID_OBJECT_ID: {
+        code: BAD_USER_INPUT,
+        type: INVALID_OBJECT_ID,
+        message: 'Provided object id does not exist, was deleted, or you do not have access to this object',
+    },
+    INVALID_SOURCE: {
+        code: BAD_USER_INPUT,
+        type: INVALID_SOURCE,
+        message: 'Provided combination of sourceType and sourceId is invalid, or you do not have rights to set this source',
+    },
+    INVALID_UNIT: {
+        code: BAD_USER_INPUT,
+        type: INVALID_UNIT,
+        message: 'Provided combination of unitName and unitType is invalid, check that provided unit belongs to the provided objectId',
+    },
+    INVALID_ADDRESS_KEY: {
+        code: BAD_USER_INPUT,
+        type: INVALID_ADDRESS_KEY,
+        message: 'Provided addressKey is wrong, check that provided addressKey belongs to the provided objectId',
+    },
+    INVALID_CUSTOM_FIELD: {
+        code: BAD_USER_INPUT,
+        type: INVALID_CUSTOM_FIELD,
+        message: 'Custom field for this value does not exist',
+    },
+    ALREADY_EXISTS_UNIQ_KEY: {
+        code: BAD_USER_INPUT,
+        type: ALREADY_EXISTS_UNIQ_KEY,
+        message: 'Another object with this uniqKey already exists in this organization',
+    },
+    ALREADY_EXISTS_OBJECT_ID: {
+        code: BAD_USER_INPUT,
+        type: ALREADY_EXISTS_OBJECT_ID,
+        message: 'Another object linked to provided customField already exists with this objectId. Get this object and update it, instead of creating a new one',
+    },
+}
+
+const CustomValue = new GQLListSchema('CustomValue', {
+    schemaDoc: 'Custom value for CustomField',
+    fields: {
+        customField: {
+            schemaDoc: 'CustomField object that this data refers to ',
+            type: 'Relationship',
+            ref: 'CustomField',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.PROTECT' },
+        },
+
+        data: {
+            schemaDoc: 'Data of this customValue',
+            type: 'Json',
+            isRequired: true,
+        },
+
+        objectId: {
+            schemaDoc: 'What instance of CustomField.schemaName this customValue relates to?',
+            type: 'Text',
+            isRequired: true,
+        },
+
+        filterDataString: {
```

Comment:
Date time and numbers

Reply:
Datetime is string, I think I'll just create a special type for that